test/field_exchanger_tests.jl

```diff
@@ -0,0 +1,192 @@
+get_field(sim::DummySimulation, ::Val{:radiative_energy_flux}) = sim.cache.radiative_energy_flux
+get_field(sim::DummySimulation, ::Val{:liquid_precipitation}) = sim.cache.liquid_precipitation
+get_field(sim::DummySimulation, ::Val{:snow_precipitation}) = sim.cache.snow_precipitation
+
+@testset "import_atmos_fields!" begin
+
+    boundary_space = TestHelper.create_space(FT)
+    coupler_names = (:F_A, :F_E, :F_R, :P_liq, :P_snow)
+    atmos_names = (
+        :turbulent_energy_flux,
+        :turbulent_moisture_flux,
+        :radiative_energy_flux,
+        :liquid_precipitation,
+        :snow_precipitation,
+    )
+    atmos_fields = NamedTuple{atmos_names}(ntuple(i -> Fields.ones(boundary_space), length(coupler_names)))
+
+    model_sims = (; atmos_sim = DummySimulation(atmos_fields))
+
+    flux_types = (CombinedAtmosGrid(), PartitionedComponentModelGrid())
+    results = [FT(1), FT(0)]
+    for (i, t) in enumerate(flux_types)
+        coupler_fields = NamedTuple{coupler_names}(ntuple(i -> Fields.zeros(boundary_space), length(coupler_names)))
+        import_atmos_fields!(coupler_fields, model_sims, boundary_space, t)
+        @test parent(coupler_fields.F_A)[1] == results[i]
+        @test parent(coupler_fields.F_E)[1] == results[i]
+        @test parent(coupler_fields.F_R)[1] == results[1]
+        @test parent(coupler_fields.P_liq)[1] == results[1]
+        @test parent(coupler_fields.P_snow)[1] == results[1]
+    end
+end
+
+# surface field exchange tests
+struct TestSurfaceSimulation1{C} <: SurfaceModelSimulation
+    cache_field::C
+end
+struct TestSurfaceSimulation2{C} <: SurfaceModelSimulation
+    cache_field::C
+end
+
+get_field(sim::Union{TestSurfaceSimulation1, TestSurfaceSimulation2}, ::Val{:surface_temperature}) =
+    sim.cache_field .* 1.0
+get_field(sim::Union{TestSurfaceSimulation1, TestSurfaceSimulation2}, ::Val{:albedo}) = sim.cache_field .* 1.0
+get_field(sim::Union{TestSurfaceSimulation1, TestSurfaceSimulation2}, ::Val{:roughness_momentum}) =
+    sim.cache_field .* 1.0
+get_field(sim::Union{TestSurfaceSimulation1, TestSurfaceSimulation2}, ::Val{:roughness_buoyancy}) =
+    sim.cache_field .* 1.0
+get_field(sim::Union{TestSurfaceSimulation1, TestSurfaceSimulation2}, ::Val{:beta}) = sim.cache_field .* 1.0
+
+get_field(sim::TestSurfaceSimulation1, ::Val{:area_fraction}) = sim.cache_field .* 0.0
+get_field(sim::TestSurfaceSimulation2, ::Val{:area_fraction}) = sim.cache_field .* 0.5
+
+@testset "import_combined_surface_fields!" begin
+    # coupler cache setup
+    boundary_space = TestHelper.create_space(FT)
+    coupler_names = (:T_S, :z0m_S, :z0b_S, :albedo, :beta)
+
+    # coupler cache setup
+    exchanged_fields = (:surface_temperature, :albedo, :roughness_momentum, :roughness_buoyancy, :beta)
+
+    sims = (; a = TestSurfaceSimulation1(ones(boundary_space)), b = TestSurfaceSimulation2(ones(boundary_space)))
+
+    flux_types = (CombinedAtmosGrid(), PartitionedComponentModelGrid())
+    results = [FT(0), FT(0.5)]
```

Comment:
I think these values come from the `area_fraction` definitions of `get_field` - could you add a comment clarifying this?

Reply:
Indeed! :)